src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -86,40 +89,84 @@ class JooqBurdenEstimateRepository(
                                   outcomeLookup: Map<String, Int>, cohortSizeId: Int,
                                   expectedDisease: String)
     {
-        val records = estimates.flatMap { estimate ->
-            if (estimate.disease != expectedDisease)
-            {
-                throw InconsistentDataError("Provided estimate lists disease as '${estimate.disease}' but scenario is for disease '$expectedDisease'")
+        val countries = getAllCountryIds()
+
+        // The only foreign keys are:
+        // * burden_estimate_set, which is the same for every row, and it's the one we just created and know exists
+        // * country, which we check below, per row of the CSV (and each row represents multiple rows in the database
+        //   so this is an effort saving).
+        // * burden_outcome, which we check below (currently we check for every row, but given these are set in the
+        //   columns and don't vary by row this could be made more efficient)
+        dsl.withoutCheckingForeignKeyConstraints(BURDEN_ESTIMATE) {
+
+            // Currently we write everything to a byte array in memory and then later feed that stream to CopyManager.
+            // Obviously this could be made more performant by chaining those two together somehow.
+            val data = ByteArrayOutputStream().use { stream ->
+                PostgresCopyWriter(stream).use { writer ->
+                    for (estimate in estimates)
+                    {
+                        if (estimate.disease != expectedDisease)
+                        {
+                            throw InconsistentDataError("Provided estimate lists disease as '${estimate.disease}' but scenario is for disease '$expectedDisease'")
+                        }
+                        if (estimate.country !in countries)
+                        {
+                            throw UnknownObjectError(estimate.country, "country")
```

Comment:
There's no test specifically for country validation, lets add one (relates to the comment on `JooqExtensions` about what happens if bad values are added)